tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
 	exit 1
 }
 
-if [[ $1 ]]; then
-	usage
-fi
+# Wrapper to check for a command silently.
+function has_command {
+	command -v "$1" &>/dev/null
+	return $?
+}
 
-# Check if we're on a Mac or Linux, bail if we're not.
-OS="$(uname)"
-if [[ "$OS" == "Linux" ]]; then
-	ON_LINUX=1
-	if ! command -v apt &>/dev/null; then
-		die "Installer script requires 'apt' to ensure essentials are installed."
+# Check if we're on Mac or Linux; bail if we're not.
+function do_system {
+	OS="$(uname)"
+	if [[ "$OS" != "Linux" && "$OS" != "Darwin" ]]; then
+		die "Installer script is only supported on macOS and Linux."
+	elif [[ "$(uname -m)" == "aarch64" ]]; then
+		# Homebrew doesn't support Linux on ARM, so gate the installer as well.
+		die "Linux on ARM is unsupported."
 	fi
-	sudo apt update
-	sudo apt install build-essential
-elif [[ "$OS" != "Darwin" ]]; then
-	die "Installer script is only supported on macOS and Linux."
-fi
+	echo "Valid OS: $OS"
+}
 
-# Check for curl and git
-if ! command -v git &>/dev/null; then
-	die "Installer script requires 'git' to be installed."
-fi
+# Checks for build-essential (Linux), git, and curl.
+function do_basics {
+	if [[ "$OS" == "Linux" ]]; then
+		if ! has_command apt; then
+			# Effectively only support Debian.
```

Comment:
Mostly we're "officially" supporting Ubuntu rather than Debian, although since I personally use Debian we wind up supporting that too in most places anyway. 😉
```suggestion
			# Effectively only support Debian, Ubuntu, and derivatives.
```

OTOH, really I wouldn't recommend using this script on Linux anyway. I'd expect Linux users would rather use apt or manually install stuff rather than using Homebrew. So the Linux support here isn't a very high priority in the first place. 🙂

If we really wanted to support Ubuntu in this script, IMO we'd skip homebrew entirely there. Bash and jq are both available from the distro, and the required version of PHP can be gotten from the https://deb.sury.org/ repo. (Debian should work with that too unless stable winds up with a too-old version of jq or something)

Reply:
I'd agree here. If we say deb/ubuntu is supported, we should give it a natural install, like what Brad mentioned. We might be able to steal from our Docker setup.

I think it's okay to note that Linux support is available but the script's audience (as of today) is really Mac users.

Reply:
> Mostly we're "officially" supporting Ubuntu rather than Debian

I really should have said "Debian-based systems".

But ultimately I agree; I only had support here because it was there previously. Chopped it altogether now: 0fac46b1f8